core/src/main/java/org/apache/calcite/interpreter/MinusNode.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.interpreter;
+
+import org.apache.calcite.rel.core.Minus;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Interpreter node that implements a
+ * {@link org.apache.calcite.rel.core.Minus}.
+ */
+public class MinusNode implements Node {
+  private final Source leftSource;
+  private final Source rightSource;
+  private final Sink sink;
+  private final Minus minus;
+
```

Comment:
`Minus` and `Intersect` has same semantics, we can reuse the node class.

Reply:
If we consider reuse the node class, should we use the same node class for `Union`, `Minus` and `Intersect`.

Reply:
Thanks for review, I'm not sure what do you mean by
> `Minus` and `Intersect` has same semantics

As far as I know, `Minus` and `Intersect` work differently.

Reply:
Minus equals to Except but different from Intersect

Reply:
Oops, my mistake, i think we can unify them into one class.

Reply:
Maybe a common abstract base class. Minus, Intersect, Union have different behavior but similar data. Put the data in the base class. Maybe some of the behavior will turn out to be common, too.

Reply:
Yes, we can use an SetOp class to unify Minus, Intersect and Union.
But the logic is quite simple, so I think we can just put them is the SetOp class.
No need more UnionNode, MinusNode or IntersectNode.